### PR TITLE
Convert rest of PCP components to Typescript

### PR DESCRIPTION
**Pull request by davish**
This PR converts the last few component files from javascript to typescript.

Redux is still purposely left untouched, since the goal is to move from redux for most server-side data fetching to [swr](https://swr.vercel.app).

### CODE REVIEW

File: frontend/plan/pages/index.tsx [hunk 1]
```diff
@@ -142,120 +142,154 @@ function Index() {
...
+                                >
+                                    <Selector mobileView={true} view={0} />
+                                </div>
+                            </div>
+                        </div>
+                        <div style={{ padding: "10px" }}>
+                            <Cart setTab={setTab} mobileView={true} />
+                        </div>
+                        <div style={{ padding: "10px" }}>
+                            {/* Unfortunately running into a weird issue with connected components and TS. */}
```
Comment (jxiao) [inline frontend/plan/pages/index.tsx:216]: Out of curiosity, what was the specific issue here?

**Review by jxiao — APPROVED** [review]
Looks good to me - just two small imports that we can remove!

File: frontend/plan/pages/index.tsx (hunk 1 above)
Comment (davish) [reply frontend/plan/pages/index.tsx:216]: honestly, no clue. Was saying a prop that was defined on the interface, wasn't actually defined. think it has to do with `connect` but can't be 100% sure

**Comment by davish** [comment]
Closing since this was merged into #270

**Pull request closed without merging**